internal/functions/gerritstatusupdater/gerritstatusupdater.go

```diff
@@ -0,0 +1,386 @@
+	"golang.org/x/oauth2"
+)
+
+var (
+	debug = os.Getenv(EnvDebug) != ""
+)
+
+// Function is a type which implements net/http.Handler, a handler which
+// implements the serverless function for handling workflow webhook events
+// and converting them into Gerrit reviews.
+type Function struct {
+	DevelopmentMode bool
+}
+
+const (
+	// EnvPrefix is used to prefix environment variables used by
+	// gerritstatusupdater. For repo-workflow-based values it is used in
+	// conjunction with the env*Suffix variables that follow.
+	//
+	// Repo-based environment variable names are constructed via
+	// buildRepoKey, where the repository name is the full_name derived
+	// from the workflow run.
+	//
+	// Repo-workflow-based environment variable names are constructed via
+	// buildRepoPathKey, where the repository name is the full_name derived from
+	// the workflow run, and the path is derived from the workflow.
+	EnvPrefix = "GERRITSTATUSUPDATER"
+
+	// envGitHubPATEnvSuffix is the suffix for the repo-based environment
+	// variable that capture the GitHub access token to use when querying for
+	// workflow run information in response to workflow job events. Note, this
+	// variable is only repo-based: it does not make sense to configure this at
+	// the workflow level (nor is it possible given the resolution of some
+	// webhook events requires further calls to the GitHub API).
+	EnvGitHubPATSuffix = "GITHUB_PAT"
+
+	// EnvGerritHubInstanceEnvSuffix is the suffix for the repo-workflow-based
+	// environment variable that captures the GerritHub instance
+	EnvGerritHubInstanceEnvSuffix = "GERRITHUB_INSTANCE"
+
+	// EnvGerritHubUsernameEnvSuffix is the suffix for the repo-workflow-based
+	// environment variable that captures the GerritHub username
+	EnvGerritHubUsernameEnvSuffix = "GERRITHUB_USERNAME"
+
+	// EnvGerritHubPasswordEnvSuffix is the suffix for the repo-workflow-based
+	// environment variable that captures the GerritHub password associated with
+	// the GerritHub username (see envGerritHubUsernameEnvSuffix).
+	EnvGerritHubPasswordEnvSuffix = "GERRITHUB_PASSWORD"
+
+	// EnvWebhookSecret is the GitHub webhooks secret that is used to sign each
+	// webhook event. The webhook secret is common to all repos because it is
+	// effectively tied to the instance of gerritstatusupdater that runs,
+	// and indeed it would be a security problem to parse the webook without
+	// first validating it.
+	EnvWebhookSecret = EnvPrefix + "WEBHOOK_SECRET"
+
+	// EnvDebug prints debug information when set to something other than the
+	// empty strying. This is useful when running this serverless function
+	// locally. When EnvDebug is set, nothing is actually written to GerritHub,
+	// instead the messages that would be sent to GerritHub are written to
+	// stdout.
+	EnvDebug = EnvPrefix + "DEBUG"
+)
+
+// ServeHTTP is the implementation of the gerritstatusupdater serverless
+// function.
+func (fun Function) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
nit: `fn` is often a better shorthand for function, and less ambiguous with the "fun" word :P

Reply:
Good call.